server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java

```diff
@@ -1316,6 +1347,8 @@ public static IndexMetaData fromXContent(XContentParser parser) throws IOExcepti
                         builder.state(State.fromString(parser.text()));
                     } else if (KEY_VERSION.equals(currentFieldName)) {
                         builder.version(parser.longValue());
+                    } else if (KEY_MAPPING_VERSION.equals(currentFieldName)) {
+                        builder.mappingVersion(parser.longValue());
```

Comment:
can we somehow make sure that if the index is 7.0 that the version is present? I really want to make sure we are not missing it.

Reply:
I pushed e1c6fe665d495abab2b30b45cd784117181f53af.

Reply:
Let me know what you think of that one @s1monw.

Reply:
Looks good